mlir/test/Ion/PulseOp.mlir

```diff
@@ -0,0 +1,71 @@
+}
+
+func.func @example_parallel_protocol_two_qubits(%arg0: f64) -> (!quantum.bit, !quantum.bit) {
+    %0 = quantum.alloc( 1) : !quantum.reg
+    %1 = quantum.extract %0[0] : !quantum.reg -> !quantum.bit
+    %2 = quantum.extract %0[1] : !quantum.reg -> !quantum.bit
+    %3:2 = ion.parallelprotocol(%1, %2): !quantum.bit, !quantum.bit {
+        ^bb0(%arg1: !quantum.bit, %arg2: !quantum.bit):
+        ion.pulse(%arg0: f64) %arg1 {beam=#ion.beam<
+        transition = #ion.transition<level_0 = #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, level_1 =#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, einstein_a=10.10>,
+        rabi=10.10,
+        detuning=11.11,
+        polarization=dense<[0, 1]>: tensor<2xi64>,
+        wavevector=dense<[0, 1]>: tensor<2xi64>>}
+        ion.pulse(%arg0: f64) %arg2 {beam=#ion.beam<
+        transition = #ion.transition<level_0 = #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, level_1 =#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, einstein_a=10.10>,
+        rabi=10.10,
+        detuning=11.11,
+        polarization=dense<[0, 1]>: tensor<2xi64>,
+        wavevector=dense<[0, 1]>: tensor<2xi64>>}
+        ion.yield %arg1, %arg2: !quantum.bit, !quantum.bit
+    }
+    return %3#0, %3#1: !quantum.bit, !quantum.bit
+}
+
+
+func.func @example_ion() -> !ion.ion {
+    %0 = ion.ion {
+    name="YB117", 
+    mass=10.1, 
+    charge=12.1, 
+    position=dense<[0, 1]>: tensor<2xi64>, 
+    levels=[#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>], 
+    transitions=[#ion.transition<level_0 = #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, level_1 =#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, einstein_a=10.10>,#ion.transition<level_0 = #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, level_1 =#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, einstein_a=10.10>]
+    }: !ion.ion
+    return %0: !ion.ion
```

Comment:
Minor: the op and the type are now both called `ion.ion`. Maybe the op can be called `createIon` or something more appropriate (I don't have enough context).